exercises/strain/canonical-data.json

```diff
@@ -0,0 +1,155 @@
+{
+  "exercise": "strain",
+  "cases": [
+    {
+      "description": "filter list returning only values that satisfy the predicate function",
```

Comment:
Where possible, we tend to try and not use nested test groups. In this case, you've used very readable descriptions for the test cases, so my preference would be to remove the nesting. Curious to hear what others think.